src/v/archival/manifest.h

```diff
@@ -145,6 +123,44 @@ class manifest final {
     model::ntp _ntp;
     model::revision_id _rev;
     segment_map _segments;
+    model::offset _last_offset;
+};
+
+class topic_manifest {
+public:
+    /// Create manifest for specific ntp
+    explicit topic_manifest(
+      const cluster::topic_configuration& cfg, model::revision_id rev);
+
+    /// Create empty manifest that supposed to be updated later
+    topic_manifest();
+
+    /// Update manifest file from input_stream (remote set)
+    ss::future<> update(ss::input_stream<char>&& is);
+
+    /// Serialize manifest object
+    ///
+    /// \return asynchronous input_stream with the serialized json
+    std::tuple<ss::input_stream<char>, size_t> serialize() const;
```

Comment:
nit: a tuple is a poor man's struct. 

I'd recommend giving this a name like `serialized_json_stream {};` but just a nit.